nft_ingester/src/program_transformers/bubblegum/update_metadata.rs

```diff
@@ -0,0 +1,292 @@
+                }
+
+                // Upsert into `asset_data` table.
+
+                let slot_i = bundle.slot as i64;
+
+                let uri = if let Some(uri) = &update_args.uri {
+                    uri.replace('\0', "")
+                } else {
+                    current_metadata.uri.replace('\0', "")
+                };
+
+                let name = if let Some(name) = update_args.name.clone() {
+                    name
+                } else {
+                    current_metadata.name.clone()
+                };
+
+                let symbol = if let Some(symbol) = update_args.symbol.clone() {
+                    symbol
+                } else {
+                    current_metadata.symbol.clone()
+                };
+
+                let primary_sale_happened =
+                    if let Some(primary_sale_happened) = update_args.primary_sale_happened {
+                        primary_sale_happened
+                    } else {
+                        current_metadata.primary_sale_happened
+                    };
+
+                let mut chain_data = ChainDataV1 {
+                    name: name.clone(),
+                    symbol: symbol.clone(),
+                    edition_nonce: current_metadata.edition_nonce,
+                    primary_sale_happened,
+                    token_standard: Some(TokenStandard::NonFungible),
+                    uses: current_metadata.uses.clone().map(|u| Uses {
+                        use_method: UseMethod::from_u8(u.use_method as u8).unwrap(),
+                        remaining: u.remaining,
+                        total: u.total,
+                    }),
+                };
+                chain_data.sanitize();
+                let chain_data_json = serde_json::to_value(chain_data)
+                    .map_err(|e| IngesterError::DeserializationError(e.to_string()))?;
+
+                let is_mutable = if let Some(is_mutable) = update_args.is_mutable {
+                    is_mutable
+                } else {
+                    current_metadata.is_mutable
+                };
+
+                let chain_mutability = if is_mutable {
+                    ChainMutability::Mutable
+                } else {
+                    ChainMutability::Immutable
+                };
+
+                unprotected_upsert_asset_data(
+                    txn,
+                    id_bytes.to_vec(),
+                    chain_mutability,
+                    chain_data_json,
+                    uri.clone(),
+                    Mutability::Mutable,
+                    JsonValue::String("processing".to_string()),
+                    slot_i,
+                    Some(true),
+                    name.into_bytes().to_vec(),
+                    symbol.into_bytes().to_vec(),
+                )
+                .await?;
+
+                // Upsert into `asset` table.
+
+                // Set base mint info.
+                let tree_id = bundle.keys.get(5).unwrap().0.to_vec();
```

Comment:
This value should be parsed from blockbuster instead of via an array index

Reply:
https://github.com/metaplex-foundation/blockbuster/pull/31

Reply:
WIll incorporate in https://github.com/metaplex-foundation/digital-asset-rpc-infrastructure/pull/134 after this PR is updated